File: C/funcRomanNumbers.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

const char* RN[] = {"I","IV","V","IX","X","XL","L","XC","C","CD","D","CM","M"};
const short VRN[]={1,4,5,9,10,40,50,90,100,400,500,900,1000};
const short TOTRN=12;

char *decimalToRoman(int number)
{
    int ind=TOTRN;
    char *retRN = NULL;
    int posRN = 0;
    while(number>0) {
      int div = number/VRN[ind];
      number = number%VRN[ind];
      while(div--) {
        /* Remember to free this area after used */
        if ( (retRN = realloc(retRN,posRN+strlen(RN[ind])+1)) == NULL ) {
          retRN=NULL;
          break;
        }
        strcpy(&retRN[posRN],RN[ind]);
        posRN+=strlen(RN[ind]);
      }
      ind--;
    }
    return retRN;
}

int romanToDecimal(char *cpRN) {
  int lenRN = strlen(cpRN);
  int retDEC = 0;
  for (int i = 0; i < lenRN; i++) {
      switch (cpRN[i]) {
          case 'I':
              retDEC += 1;
              break;
          case 'V':
              retDEC += 5;
              break;
          case 'X':
              retDEC += 10;
              break;
          case 'L':
              retDEC += 50;
              break;
          case 'C':
              retDEC += 100;
              break;
          case 'D':
              retDEC += 500;
              break;
          case 'M':
              retDEC += 1000;
              break;
          default:
              // Character not recognized
              return -1;
      }

      if ( i > 0 ) {

         if ((cpRN[i] == 'V' || cpRN[i] == 'X') && cpRN[i - 1] == 'I') {
            retDEC -= 1 + 1;
         }
 
         if ((cpRN[i] == 'L' || cpRN[i] == 'C') && cpRN[i - 1] == 'X') {
            retDEC -= 10 + 10;
         }

         if ((cpRN[i] == 'D' || cpRN[i] == 'M') && cpRN[i - 1] == 'C') {
            retDEC -= 100 + 100;
         }
      }
  }
  return retDEC;
}
/* 
TO TEST
void *main(int argc, char**argv) {
  printf("\n Roman Number %s : %d\n",argv[1],romanToDecimal(argv[1]));
  printf("\n Roman Number %s : %d\n",argv[1],romanToDecimal2(argv[1]));
}	
*/

File: JAVASCRIPT/funcRomanNumbers.js
function decimalToRoman( number ) {
  const RN = ['I','IV','V','IX','X','XL','L','XC','C','CD','D','CM','M'];
  const VRN =[1,4,5,9,10,40,50,90,100,400,500,900,1000];
  var div=0;

  var ind = RN.length-1;
  var retRN = '';

  while ( number > 0 ) {
	  div = Math.floor(number/VRN[ind]);
	  number=number%VRN[ind];

	  while ( div-- ) {
	    retRN = retRN + RN[ind];
  	}
	  ind--;
  }

  return retRN;
}

function romanToDecimal( RN ) {
  var retDEC = 0;
  for (let i = 0; i < RN.length; i++) {

      switch (RN.charAt(i)) {
             case 'I':
		  retDEC=retDEC+1;
                  break;
             case 'V':
		  retDEC=retDEC+5;
                  break;
             case 'X':
		  retDEC=retDEC+10;
                  break;
             case 'L':
		  retDEC=retDEC+50;
                  break;
             case 'C':
		  retDEC=retDEC+100;
                  break;
             case 'D':
		  retDEC=retDEC+500;
                  break;
             case 'M':
		  retDEC=retDEC+1000;
                  break;
             default:
		  return -1;
      }
      if ( i ) {
         if ((RN.charAt(i) === 'V' || RN.charAt(i) === 'X') && RN.charAt(i - 1) === 'I') {
            retDEC -= 1 + 1;
         }
         if ((RN.charAt(i) === 'L' || RN.charAt(i) === 'C') && RN.charAt(i - 1) === 'X') {
            retDEC -= 10 + 10;
         }
         if ((RN.charAt(i) === 'D' || RN.charAt(i) === 'M') && RN.charAt(i - 1) === 'C') {
            retDEC -= 100 + 100;
         }
      }
  }

  return retDEC;
}

